### DISCUSSION TITLE
Customizing app or vendor js/css by page

### DISCUSSION THREAD

**Discussion by JeffSpies** [quasarframework/quasar#10600] [General - Components / Directives / etc]
Hi, all,

I have a somewhat silly use case. It's an app with external and internal pages.

- The publically accessible external pages need to meet a certain Google Lighthouse/[PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/) score
- The auth-only internal pages, where all the complexity is, do not need to meet any lighthouse requirements, and, e.g., the longer page loads aren't a big deal

Much of the issue is due to "unused code" in the app and vendor files:

- I'm using the `()=>import(page)` style routing
- I've disabled vendor in `quasar.conf`
- I've started removing otherwise convenient boot files to get my app files smaller
- SSR didn't do much as the bulk of the issues reside in the unused code

So while I'll likely meet the Lighthouse requirements, I have a less efficient and less convenient to develop the internal app all because of a few external pages.

Without having to create two apps, is there a way to get more granular about what is loaded in app and/or vendor by page? I suspect that this may be more of a webpack issue, but if anyone has similar examples I could peruse, I'd appreciate it!

Thanks!

**Comment by ontwikkelfabriek** [comment]
@JeffSpies 

I would take a look at these videos/source code. It will reduce the bundle size of a clean newly created(with CLI) Quasar app by +-70%. I reduced it even further by using a compression plugin and moving the generated sets to the assets folder instead of the public.  See the youtube conversation in Part2 between Michael v/d Dobbel (that's me) and Danny.


> The publically accessible external pages need to meet a certain Google Lighthouse/PageSpeed Insights score

I had a big improvement in Lighthouse performance score. 

Part1:
https://www.youtube.com/watch?v=WWWEA9UGY3A

Part2 ( with the discussion how to decrease the bundle size even further).
https://www.youtube.com/watch?v=na17MMNHwkM

Source code ( without the discussed optimizations):
https://www.dconnell.co.uk/#/resource/quasar-icon-library-source-code

Of course there a tons of other ways to increase Lightning scores (in Quasar).
A lot of them are used by this Quasar project, it has the highest Lighthouse scores I have seen for a Quasar App:
https://github.com/stelace/marketplace-demo

See here for a summary of what they did to increase the performance:
https://github.com/stelace/marketplace-demo#fast-by-default-checkered_flag
